certbot/compat/path.py

```diff
@@ -0,0 +1,31 @@
+"""This compat module wraps os.path to forbid some functions."""
```

Comment:
Do we want to let people import this module directly? I see no reason why we should.

What do you think about calling this file `_path.py`? We haven't done anything like this in Certbot yet, although I think we should as part of a resolution of https://github.com/certbot/certbot/issues/5775.

Reply:
Using underscored name modules for internal uses was already in my suggestion list for the team. I will gladly rename the module.

Reply:
Done.

Reply:
I'd love to see that suggestion list sometime!